### ISSUE TITLE
TES3MP no longer launches after SteamOS updates.

### ISSUE THREAD

**Issue by husky3g** [jefetienne/tes3mp-container-install#1]
I've been having this issue a few times but never to this extent - the exact same error happened before previously but a combination of tinkering/reinstalling seemed to resolve the issue and I pushed it off and procrastinated bothering with it because TES3MP started working again.

Seemingly, any time Valve updates certain things in SteamOS, it breaks the launch script created by the installer.

`localuser:deck being added to access control list
xterm: cannot load font "-misc-fixed-medium-r-semicondensed--13-120-75-75-c-60-iso10646-1"
`

is what comes up in konsole when attempting to run the launch script located at "/home/deck/Games/tes3mp/tes3mp-client.sh"

Again, this issue happened for me before, but a combination of randomly trying to install xterm fonts and Google-fu and experimentation fixed the problem. I believe I also completely deleted TES3MP from /home/deck/Games/tes3mp/ and reinstalled and that may have resolved the issue.

If I try to run the tes3mp shell script directly in konsole I get the following:

`Loading client config from the package directory
./tes3mp.x86_64: error while loading shared libraries: libzvbi.so.0: cannot open shared object file: No such file or directory
`

**Comment by husky3g** [comment]
This is the message I get when attempting to reinstall - I click "yes" to install TES3MP, (it skips the download since the folder exists), then it asks if I want to update Distrobox, I click "no", it runs, gives the below message, then asks if I want to run the TES3MP setup wizard, I select "yes" then it just opens the OpenMW wizard and then closes when finished with OpenMW wizard and nothing happens. When I attempt to launch, I get the initial error  I posted.

![Screenshot](https://github.com/jefetienne/tes3mp-container-install/assets/70312839/60271e3d-9f2b-4229-a1f0-3647eccc0a55)

**Comment by husky3g** [comment]
If I follow the same process and select "Yes" to update Distrobox, I get this and the launcher still gives the same xterm error.

![Screenshot2](https://github.com/jefetienne/tes3mp-container-install/assets/70312839/a2e7bc9c-898b-4922-a085-bfa2fe84556e)

**Comment by jefetienne** [comment]
Hey there! Thanks for bringing up the issue.

Other than `xterm: cannot load font "-misc-fixed-medium-r-semicondensed--13-120-75-75-c-60-iso10646-1"` which I believe is a harmless error, I wasn't able to reproduce your issue. However, your screenshot showed me this part:

```
Installing latest version of Podman...
This script is deprecated and unsupported
head over to:

https://www.github.com/...
```

It turns out the install script is no longer updating distrobox properly, so I updated the install script. Can you try using the latest version on the main branch and see if it works?

Furthermore, it looks like part of the issue you might be getting is that apparently SteamOS 3.5+ actually includes a built-in /usr/bin/distrobox and podman, but I'm failing the ability to enter containers with it. I'm forcing the TES3MP shell scripts to force using $HOME/.local/bin/distrobox until I get that working.

If you could test for me by confirming `command -v distrobox` gives you /usr/bin/distrobox and try this:

```
distrobox-create --image ubuntu:22.04
distrobox enter ubuntu-22-04
```

I'm getting this, I'm wondering if you are too:
```
Error: unable to start container "620cb74bf5b931ffbb87f7b7f8b4fdebced7e8ef4054e926ad377d53e79d412f": exit status 1
```

**Comment by husky3g** [comment]
Just downloaded the updated version it DOES launch now. The konsole window with the xterm error stays open when closing out the tes3mp client but otherwise it works. Since the gaming UI closes "all opened windows" it closes the console window automatically when exiting the game, it's a non-issue, it seems to only stay open after closing TES3MP when in desktop mode. Not a big deal, just mentioning because it doesn't seem to be functioning as intended. I appreciate the fix though! 

Should make it easier next time Valve decides to nuke my TES3MP. I'll let you close this - I'm not well versed on Github so I'll leave it to you unless it's something I have to do. Thanks again!

scratch that. restarted my Deck, now getting this....

![Screenshot_20240305_005027](https://github.com/jefetienne/tes3mp-container-install/assets/70312839/5d0b9091-3d03-45f6-8c10-48733617f977)

When I hit Y, Enter it clses and nothing happens. konsole still open in the background.

**Comment by husky3g** [comment]
Ran the install again, getting this:

![Screenshot_20240305_005027](https://github.com/jefetienne/tes3mp-container-install/assets/70312839/93ae9b86-1ff7-4d85-84c3-0d0984763a39)

**Comment by husky3g** [comment]
So I DID get it working again, I renamed the podman-static folder under deck/local/share to podman-static.bak and let it run the installer again. The game launches  and runs now, but it very frequently crashes now. It may be because of Tamriel Rebuilt but I'm not positive. I haven't gotten any crashes playing it on my Windows desktop playing it, only on my Deck so again, not sure.

**Comment by jefetienne** [comment]
I'm glad you got it fixed up! You might be able to get some logs regarding the crashes by modifying [this line](https://github.com/jefetienne/tes3mp-container-install/blob/032589fce66f929e3e450a654323584f527432bb/tes3mp-container-install.sh#L155) in your $HOME/Games/tes3mp/tes3mp-client.sh to include `; exec bash` after so the xterm doesn't close out after tes3mp exits.

```diff
- xterm -e bash -c '$distroboxDir/distrobox-enter $containerName -- $installpath/TES3MP/tes3mp'
+ xterm -e bash -c '$distroboxDir/distrobox-enter $containerName -- $installpath/TES3MP/tes3mp; exec bash'
```

**Comment by husky3g** [comment]
Hmm. I'm getting the "podman system migrate" error again with the y/n prompt. Not sure what's going on. Any way to send you some diagnostic info?

Any way to clear out all of the files and fresh install?

**Comment by husky3g** [comment]
Once again, got it running again by opening the share folder and opening konsole in Dolphin with F4, doing `sudo rm -r podman-static` and deleting the other .bak folders again and re-running the install.

I'm hoping that it doesn't do the same again. Will try to change the xterm command and report back. Thanks for the help.

**Comment by jefetienne** [comment]
I started noticing I've been getting it too. I'm going to have to see what is going on and if it's only reproducible on Game Mode, etc.

I may also look into the option of having a log file be written to as the client or server is being run. I'm pretty sure you can do two processes at once in bash.

Thanks for staying on top of it. I'm happy there's someone out there using this project :)

**Comment by husky3g** [comment]
No problem! Been religiously playing my Deck and TES3MP so this installer has been a lifesaver. Still working, but getting frequent crashes. Happens on my desktop too on Nerevarine Prophecies so I assume it may not be related to the install.

**Comment by husky3g** [comment]
Was working for a week, back to the last screenshot i posted. Nerevarine Prophecies server was causing the crashes, because it frequently happens on my desktop as well, so the main issue is the install randomly stopping from launching once again.

**Comment by jefetienne** [comment]
I made another update to the shell script, this time forcing the use of your root installation and skipping installing it locally due to the breaking changes of 1.4.2 and 1.7.0 + podman-launcher path and config differences. Can you try:
1. Removing
    - `~/.local/share/podman-static`
    - `~/.local/bin/distrobox*` (all bins with `distrobox`)
    - `~/.local/podman`
    - `~/.config/containers`
2. Running the latest script?

I was able to get it running!

**Comment by husky3g** [comment]
Sorry, been a bit busy. Docked my Deck with a keyboard/mouse right now - going to try running delete commands on those folders in konsole and then retrying it. Will edit this comment with results after reinstall.

Quick update - 
```
    ~/.local/podman
    ~/.config/containers
```
Neither folder existed on my Deck. (TES3MP hasn't been running, so it's possible they were removed by an update or something) but I was able to remove all bins for Distrobox and delete the podman-static with "sudo rm -r podman-static" - For whatever reason, I am/was not able to delete the folder without sudo.

I started the install after everything was deleted, then the installer was stuck on "setting up Distrobox container" and didn't stop for about 30 minutes. I closed it, restarted, seems to be the same.

![Screenshot_20240322_235838](https://github.com/jefetienne/tes3mp-container-install/assets/70312839/c2e988aa-202a-4570-80ca-f0f00746fea4)

EDIT2: After restarting my Steam Deck after deleting everything, tried the install, ran perfect. (Typical crashes from playing on Nerevarine Prophecies (happens on Windows as well) but otherwise it is good. 

Still getting that xterm font error in the terminal window when the game is running and the window has to be closed manually after the game closes. Still not sure why that's happening but it affects nothing.

**Comment by jefetienne** [comment]
I'm really glad it's working for you! I'll close the issue but will reopen if you have any other future problems with starting the client/server back up, but I'm happy it's been resolved :) I myself play from time-to-time so if I encounter other issues, I'll be making updates. Maybe even start making official "releases" for notifications.

EDIT: Released 1.0.0!
